pairing/pairing.md

```diff
@@ -0,0 +1,442 @@
+
+For in person *pairs*:
+
+Able to see each other well enough to pick up on visual cues
+
+For remote *pairs*:
+
+* Both *pairers* need to be able to see each other while working, this can be tricky when sharing a screen.
+* The *pairers* should be able to see each other's visual cues.
+
+### Whiteboard
+
+Having a whiteboard or some paper available for diagrams can help facilitate discussion immensely.
+
+For in person *pairs*, an actual whiteboard or some paper works well.
+
+For remote *pairs*, a piece of paper with the camera trained on it, or even better some virtual whiteboard.
+
+## Roles
+
+The two roles listed above, *driver* and *navigator* serve different purposes. It's often not clear who does what, keeping the guidelines here in mind can be helpful.
+
+### *Driver*
+
+This is the clearest role, at least in part, because it's the person at the keyboard. The *driver* is responsible for the implementation. They keep their focus on what is happening right now and interpret what the *navigator* is saying. It's best if the *driver* doesn't focus too much on the broader design. Instead they should focus on making sure the work they are doing right now is high quality and doesn't contain errors. For example, while creating a new class as directed by the *navigator*, the *driver* would focus on:
+
+* extracting variables
+* extracting private methods
+* variable, method, and class names
+* other local refactoring and improvements.
+* method level logic
+* private methods
+* code style
+* running the tests
+
+### *Navigator*
+
+This role focuses on the broader scope of the *problem*. Generally the *navigator* will set the direction the code should go in. For example they might say something like "Maybe we can pull these shared methods into another class and pass it into the constructors". The *driver* will then take this and put it into code, modifying it to keep the code clean and error free. Along the way, the *navigator* is responsible for keeping the *pairing* on track and ensuring that rabbit holes are stepped out of in a reasonable time or that new approaches are tried as old ones fail to pan out. The *navigator* is also responsible to keeping an eye out for any errors, typos, or refactoring opportunities that crop up and the *driver* misses. Though this is not their primary role it's still important.
+
+### Dialogue
+
+It's important to note that the *driver* isn't just implementing what the *navigator* says, they are part of a dialogue. The idea here is that the *driver* brings in the "low level" perspective from the code that's in front of them and what will actually work here, the *navigator* brings in a more bird's eye view and tries to imagine how what is being written now will fit in with the greater design. By switching roles with some frequency, the *pair* can ensure that they have a good handle of both perspectives. 
```

Comment:
While I think this is likely true, I can't speak with confidence about it. Frankly I haven't done enough research into why pairing works to feel fully confident in what I'm saying. This doc is also more about describing how to pair and what the expected benefits are without stepping too far into the details of why it works (I wet my toes a little since it's important to explain some stuff).

Reply:
ok